src/components/ChecSlideoutPanel.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <transition
+    appear
+    name="panel"
+    :duration="500"
+  >
+    <div class="slideout-panel">
+      <Overlay
+        v-if="renderOverlay"
+        :overlay="overlay"
+        :close-on-overlay-click="closeOnOverlayClick"
+        @close="emitClose"
+      />
+      <Panel
+        v-if="panelOpen"
+        :title="title"
+        :title-tag="titleTag"
+        :width="width"
+        :size="size"
+        :depth="depth"
+        @close="emitClose"
+      >
+        <slot />
+        <template #toolbar>
+          <slot name="toolbar" />
+        </template>
+      </Panel>
+    </div>
+  </transition>
+</template>
+
+<script>
+import Panel from './ChecSlideoutPanel/Panel';
+import Overlay from './ChecSlideoutPanel/Overlay';
+
+const panels = [];
+
+export default {
+  name: 'ChecSlideoutPanel',
+  components: {
+    Panel,
+    Overlay,
+  },
+  props: {
+    /**
+     * The size of the panel width. One of "half", "third", "threeQuarters", "full".
+     */
+    size: {
+      type: String,
+      default: 'half',
+    },
+    /**
+     * The title of the panel slideout in the header
+     */
+    title: {
+      type: String,
+      required: true,
+    },
+    /**
+     * The HTML tag to use for the title of the panel header
+     */
+    titleTag: {
+      type: String,
+      default: () => 'h2',
+    },
+    /**
+     * Controls the max width of the content in the panel. Use one of the tailwind sizes, e.g. md, xl, 2xl.
+     */
+    width: {
+      type: String,
+      default: '2xl',
+    },
+    /**
+     * The overlay backdrop when the panel is opened. One of "light" and "dark".
+     */
+    overlay: {
+      type: String,
+      default: 'light',
+      validator(value) {
+        return ['light', 'dark'].includes(value);
+      },
+    },
+    /**
+     * Whether to close panel on overlay click. Default is true.
+     */
+    closeOnOverlayClick: Boolean,
+  },
+  data() {
+    return {
+      panelOpen: true,
+      renderOverlay: true,
+      // eslint-disable-next-line vue/no-unused-properties
+      depth: 0,
+    };
+  },
+  mounted() {
+    if (panels.length) {
+      this.renderOverlay = false;
+      panels.forEach((panel) => {
+        // eslint-disable-next-line no-param-reassign
+        panel.depth += 1;
+      });
+    }
+    panels.push(this);
+    console.log('Added', panels);
+  },
```

Comment:
I'm not too familiar just yet with how you've implemented this, but when I imagined it I was thinking that the panels don't need to know about each other. The "depth" prop is fine though, because you want to tell the panel how indented it should be (e.g. when it overlays another panel it has a depth of 2, etc). Any more logic than a number prop being used to add a CSS class feels like overkill to me. What do you think?

Reply:
Thinking more about this and chatting about it, it does actually make sense to do it this way. Some comments might help to clarify exactly why things are happening e.g. why we use an external variable over data property state.